subxt/src/rpc.rs

```diff
@@ -135,6 +135,85 @@ impl From<NumberOrHex> for BlockNumber {
     }
 }
 
+impl Default for NumberOrHex {
+    fn default() -> Self {
+        Self::Number(Default::default())
+    }
+}
+
+impl NumberOrHex {
+    /// Converts this number into an U256.
+    pub fn into_u256(self) -> U256 {
+        match self {
+            NumberOrHex::Number(n) => n.into(),
+            NumberOrHex::Hex(h) => h,
+        }
+    }
+}
+
+impl From<u32> for NumberOrHex {
+    fn from(n: u32) -> Self {
+        NumberOrHex::Number(n.into())
+    }
+}
+
+impl From<u64> for NumberOrHex {
+    fn from(n: u64) -> Self {
+        NumberOrHex::Number(n)
+    }
+}
+
+impl From<u128> for NumberOrHex {
+    fn from(n: u128) -> Self {
+        NumberOrHex::Hex(n.into())
+    }
+}
+
+impl From<U256> for NumberOrHex {
+    fn from(n: U256) -> Self {
+        NumberOrHex::Hex(n)
+    }
+}
+
+/// An error type that signals an out-of-range conversion attempt.
+pub struct TryFromIntError(pub(crate) ());
```

Comment:
```suggestion
#[derive(thiserror::Error)]
pub struct TryFromIntError(pub(crate) ());
```
Then will integrate nicely with error libs and `?`. Also maybe some way to not swallow the actual inner error here, which I assumes would be an overflow.